libnest/src/package/manifest.rs

```diff
@@ -1,23 +1,86 @@
 //! Provides `PackageManifest`, a type representing a package's metadata and dependencies.
 
-use crate::package::{CategoryName, Kind, PackageName, RepositoryName, Tag};
-use semver::Version;
+use std::collections::HashMap;
 
-/// Represents a package's metadata, like its name, category, description etc.
-#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Default)]
+use chrono::{DateTime, Utc};
+use semver::{Version, VersionReq};
+use serde_derive::{Deserialize, Serialize};
+
+use super::identification::PackageFullName;
+use super::{CategoryName, Kind, PackageName};
+
+/// Represents a package's metadata, like its name, category, description, etc.
+#[derive(Serialize, Deserialize, Clone, Eq, PartialEq, Hash, Debug)]
 pub struct Metadata {
     name: PackageName,
     category: CategoryName,
-    repository: RepositoryName,
+    version: Version,
+    #[serde(default)]
     description: String,
-    tags: Vec<Tag>,
+    #[serde(default)]
+    tags: String,
```

Comment:
Not a request for modifications, but on the long term, would be great to switch this to a `Vec<Tag>`, where tag a strong type over String.